command/agent/command.go

```diff
@@ -76,6 +76,10 @@ func (c *Command) readConfig() *Config {
 	cmdFlags.Var((*AppendSliceValue)(&tags), "tag",
 		"tag pair, specified as key=value")
 	cmdFlags.StringVar(&cmdConfig.Discover, "discover", "", "mDNS discovery name")
+	cmdFlags.Var((*AppendSliceValue)(&cmdConfig.SRVRecords), "srvrecord",
```

Comment:
Thinking about this, would WAN flavors make sense, too?

I think I'd go with `join-srv` and `retry-join-srv` (and `-wan` variants) so it has good symmetry with the other `join` forms. These would go into different buckets but processed in the same way. I think you could create a single new function based on what you have and pass it the bucket of SRV records to try, and a pointer to a function that handles the LAN/WAN and retry details so we don't have to write a bunch of variants like the current `Command.retryJoin` and `Command.retryJoinWan` are.

If you did this you could respect the existing retry interval and max attempts variables, too.